Open62541.xs

```diff
@@ -795,6 +828,46 @@ UA_Client_connect(client, endpointUrl)
 	OPCUA_Open62541_Client		client
 	char *				endpointUrl
 
+OPCUA_Open62541_StatusCode
+UA_Client_connect_async(client, endpointUrl, callback, data)
+	OPCUA_Open62541_Client		client
+	char *				endpointUrl
+	SV *  				callback
+	SV *				data
+    CODE:
+	if (! SvOK(callback)) {
+		/* ignore callback and data if no callback is defined */
+		RETVAL = UA_Client_connect_async(client, endpointUrl, NULL,
+		    NULL);
+	} else {
+		if (! SvROK(callback))
+			croak("callback is not a reference");
+		if (SvTYPE(SvRV(callback)) != SVt_PVCV)
+			croak("callback is not a code reference");
+
+		PerlClientCallback *pcc = malloc(sizeof(PerlClientCallback));
```

Comment:
Could you put the declaration into an INIT: block?  In traditional C you declare at the begingin PerlClientCallback *pcc;